### DISCUSSION TITLE
WaterdogPE permissions

### DISCUSSION THREAD

**Discussion by JohnnyX071** [WaterdogPE/WaterdogPE#77] [Q&A]
Hello, I just started using waterdogPE and I want to know what is the permission that I need to give to the players to be able to use /transferserver in pocketmine?

**Comment by Alemiz112** [comment]
Hi, the `/transferserver` command is part o the pmmp. It is not proxy command. 
You probably mean `/server` which comes from proxy. By default the permission for this command is `waterdog.player.transfer`. Remember that this command is not known to downstream and therefore only player itself can execute it. 

The default permissions are defined in [`lang.ini`](https://github.com/WaterdogPE/WaterdogPE/blob/master/src/main/resources/lang.ini) file of the proxy, you can adjust or find more permissions assigned to commands.

**Reply by JohnnyX071** [reply]
> Hi, the `/transferserver` command is part o the pmmp. It is not proxy command.
> You probably mean `/server` which comes from proxy. By default the permission for this command is `waterdog.player.transfer`. Remember that this command is not known to downstream and therefore only player itself can execute it.
> 
> The default permissions are defined in [`lang.ini`](https://github.com/WaterdogPE/WaterdogPE/blob/master/src/main/resources/lang.ini) file of the proxy, you can adjust or find more permissions assigned to commands.

Oh, so how can I do it so when a player presses an npc he gets transferred to another server inside the proxy?

**Reply by Alemiz112** [reply]
It is possible to use and send `TransferPacket` to the player. Proxy will catch it and if it will match proxy known downstream server, player will be transfererd.
In the `TransferPacket` set the new donwstream name as `address`  attribute.

**Reply by JohnnyX071** [reply]
And how do I do that? Sorry but I am very new. I though I could give the perms to a player  to transfer and put in the npc to rca the player to transfer. Can I do that?

**Reply by Alemiz112** [reply]
You would need to create own pmmp plugin. Than the transfer method would like as the following:
```php
function transferPlayer(Player $player, string $serverName) : void {
	$packet = new TransferPacket();
	$packet->address = $serverName;
	$packet->port = 12345; // Random port (it doesnt matter)
	$player->dataPacket($packet);
}
```

**Reply by JohnnyX071** [reply]
So there is not another way to transfer players with an npc without needing a plugin?

**Reply by Alemiz112** [reply]
No. If you find it difficult to code own plugin, you might try to ask on our Discord server if someone has already made some public library/plugin.

**Reply by JohnnyX071** [reply]
Okay, thanks!!

**Reply by Wertzui123** [reply]
You should also be able to use /transferserver for this purpose.
The player will need the `pocketmine.command.transferserver` permission though and will also be able to use /transferserver without any NPCs.
Then you can set the command of the NPC to `rca {player} transferserver Subserver` and it should work.

**Reply by royaljacques** [reply]
use slapper , and /slapper edit
